beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -55,39 +55,134 @@ func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	headState, err := s.ChainInfoFetcher.HeadStateReadOnly(ctx)
+	if err != nil {
+		httputil.HandleError(w, "Could not get head state: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	attestations := s.AttestationsPool.AggregatedAttestations()
+	unaggAtts, err := s.AttestationsPool.UnaggregatedAttestations()
+	if err != nil {
+		httputil.HandleError(w, "Could not get unaggregated attestations: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	attestations = append(attestations, unaggAtts...)
+
+	isEmptyReq := rawSlot == "" && rawCommitteeIndex == ""
+	filteredAtts := make([]interface{}, 0, len(attestations))
+
 	for _, att := range attestations {
-		committeeIndexMatch := rawCommitteeIndex != "" && att.GetData().CommitteeIndex == primitives.CommitteeIndex(committeeIndex)
-		slotMatch := rawSlot != "" && att.GetData().Slot == primitives.Slot(slot)
-		shouldAppend := (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch))
-		if shouldAppend {
-			a, ok := att.(*eth.Attestation)
-			if ok {
-				filteredAtts = append(filteredAtts, structs.AttFromConsensus(a))
-			} else {
-				httputil.HandleError(w, fmt.Sprintf("unable to convert attestations of type %T", att), http.StatusInternalServerError)
+		var includeAttestation bool
+		if headState.Version() >= version.Electra {
+			attElectra, ok := att.(*eth.AttestationElectra)
+			if !ok {
+				httputil.HandleError(w, fmt.Sprintf("Unable to convert attestation of type %T", att), http.StatusInternalServerError)
+				return
+			}
+			data := attElectra.GetData()
+			attStruct := structs.AttElectraFromConsensus(attElectra)
+
+			includeAttestation = shouldIncludeAttestation(data, rawSlot, slot, rawCommitteeIndex, committeeIndex, isEmptyReq)
+			if includeAttestation {
+				filteredAtts = append(filteredAtts, attStruct)
+			}
+		} else {
+			attOld, ok := att.(*eth.Attestation)
+			if !ok {
+				httputil.HandleError(w, fmt.Sprintf("Unable to convert attestation of type %T", att), http.StatusInternalServerError)
 				return
 			}
+			data := attOld.GetData()
+			attStruct := structs.AttFromConsensus(attOld)
+
+			includeAttestation = shouldIncludeAttestation(data, rawSlot, slot, rawCommitteeIndex, committeeIndex, isEmptyReq)
+			if includeAttestation {
+				filteredAtts = append(filteredAtts, attStruct)
+			}
+		}
+	}
+
+	attsData, err := json.Marshal(filteredAtts)
+	if err != nil {
+		httputil.HandleError(w, "Could not marshal attestations: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	httputil.WriteJson(w, &structs.ListAttestationsResponse{
+		Version: version.String(headState.Version()),
+		Data:    attsData,
+	})
+}
+
+// Helper function to determine if an attestation should be included
+func shouldIncludeAttestation(
+	data *eth.AttestationData,
+	rawSlot string,
+	slot uint64,
+	rawCommitteeIndex string,
+	committeeIndex uint64,
+	isEmptyReq bool,
+) bool {
+	if isEmptyReq {
+		return true
```

Comment:
This part is not really needed. If both are empty, then `return committeeIndexMatch && slotMatch` will return `true`. Also you will be able to have one function argument less.